### ISSUE TITLE
Loss appears nan when training whu dataset

### ISSUE THREAD

**Issue by ljb-1** [walking-shadow/Official_Remote_Sensing_Mamba#17]
![微信图片_20240424000149](https://github.com/walking-shadow/Official_Remote_Sensing_Mamba/assets/76945920/b053b034-3559-45ba-a108-3cec5f3f4959)
Why does the boss appear to be nan in the 4th round of training?

**Comment by walking-shadow** [comment]
The nan may be due to half precision issues or the gradient clip opeartion, you can cancel amp and use full precision training, or use the code `torch.nn.utils.clip_grad_norm_(net.parameters(), 1, norm_type=2)` when back propogation may also help.

**Comment by LiuPengJX** [comment]
> he nan may be due to half precision issues or the gradient clip opeartion, you can cancel amp and use full precision training, or use the code `torch.nn.utils.clip_grad_norm_(net.parameters(), 1, norm_type=2)` when back propogation may also help.
I have encountered the same problem. Have you solved it according to the author's method?

**Comment by LPeng625** [comment]
> ![微信图片_20240424000149](https://private-user-images.githubusercontent.com/76945920/324907966-b053b034-3559-45ba-a108-3cec5f3f4959.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNzQwNzIsIm5iZiI6MTcxNDM3Mzc3MiwicGF0aCI6Ii83Njk0NTkyMC8zMjQ5MDc5NjYtYjA1M2IwMzQtMzU1OS00NWJhLWExMDgtM2NlYzVmM2Y0OTU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDA2NTYxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNTNhNGRiZDk2ZTliYjA0Nzg2Y2NhNTJkNDgzZTdiYjM3NzJkYWVkOGY2ZmU2NzM2OWI4MDM4MjYyMGJhNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CC2BvLAbqwBdO-jvMaxc5YseoD7Z-oWI64C3_Mpv15Y) Why does the boss appear to be nan in the 4th round of training?

I have encountered the same problem. Have you solved it according to the author's method?

**Comment by blacksheeplzu** [comment]
> > ![微信图片_20240424000149](https://private-user-images.githubusercontent.com/76945920/324907966-b053b034-3559-45ba-a108-3cec5f3f4959.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNzQwNzIsIm5iZiI6MTcxNDM3Mzc3MiwicGF0aCI6Ii83Njk0NTkyMC8zMjQ5MDc5NjYtYjA1M2IwMzQtMzU1OS00NWJhLWExMDgtM2NlYzVmM2Y0OTU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDA2NTYxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNTNhNGRiZDk2ZTliYjA0Nzg2Y2NhNTJkNDgzZTdiYjM3NzJkYWVkOGY2ZmU2NzM2OWI4MDM4MjYyMGJhNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CC2BvLAbqwBdO-jvMaxc5YseoD7Z-oWI64C3_Mpv15Y) Why does the boss appear to be nan in the 4th round of training?
> 
> I have encountered the same problem. Have you solved it according to the author's method?

![image](https://github.com/walking-shadow/Official_Remote_Sensing_Mamba/assets/102464114/58021e43-9475-4b43-ab5d-22244a1096ae)
it seems did not work ，I used both

**Comment by FhilMs** [comment]
I recently came across the same issue while training / validating on my own dataset, the loss became NAN at about 80 epoches.However, the code works fine on the other dataset. Eventually I found that there is something wrong with my val dataset(the number of images is different with the number of labels) And I did not add the code "torch.nn.utils.clip_grad_norm_(net.parameters(), 1, norm_type=2)" So you may check your dataset.

**Comment by FhilMs** [comment]
> > > ![微信图片_20240424000149](https://private-user-images.githubusercontent.com/76945920/324907966-b053b034-3559-45ba-a108-3cec5f3f4959.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNzQwNzIsIm5iZiI6MTcxNDM3Mzc3MiwicGF0aCI6Ii83Njk0NTkyMC8zMjQ5MDc5NjYtYjA1M2IwMzQtMzU1OS00NWJhLWExMDgtM2NlYzVmM2Y0OTU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDA2NTYxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNTNhNGRiZDk2ZTliYjA0Nzg2Y2NhNTJkNDgzZTdiYjM3NzJkYWVkOGY2ZmU2NzM2OWI4MDM4MjYyMGJhNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CC2BvLAbqwBdO-jvMaxc5YseoD7Z-oWI64C3_Mpv15Y) Why does the boss appear to be nan in the 4th round of training?
> > 
> > 
> > I have encountered the same problem. Have you solved it according to the author's method?
> 
> ![image](https://private-user-images.githubusercontent.com/102464114/326633928-58021e43-9475-4b43-ab5d-22244a1096ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ5NzQwNjQsIm5iZiI6MTcxNDk3Mzc2NCwicGF0aCI6Ii8xMDI0NjQxMTQvMzI2NjMzOTI4LTU4MDIxZTQzLTk0NzUtNGI0My1hYjVkLTIyMjQ0YTEwOTZhZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwNlQwNTM2MDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zOWUwZjZjOTdjZThmZWIyYmFlYzE2YjNhZmZlYjI1NDAzODNiMzk2ZDZhZDE3NDE2ZTg4ZjIxNDUzNGZkN2U1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6K_gdiGHSQRar2C_JripNm8ENS_OX2QlTfLPNAOXv4o) it seems did not work ，I used both

I recently came across the same issue while training / validating on my own dataset, the loss became NAN at about 80 epoches.However, the code works fine on the other dataset. Eventually I found that there is something wrong with my val dataset(the number of images is different with the number of labels) And I did not add the code "torch.nn.utils.clip_grad_norm_(net.parameters(), 1, norm_type=2)" So you may check your dataset.

**Comment by FhilMs** [comment]
> > ![微信图片_20240424000149](https://private-user-images.githubusercontent.com/76945920/324907966-b053b034-3559-45ba-a108-3cec5f3f4959.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNzQwNzIsIm5iZiI6MTcxNDM3Mzc3MiwicGF0aCI6Ii83Njk0NTkyMC8zMjQ5MDc5NjYtYjA1M2IwMzQtMzU1OS00NWJhLWExMDgtM2NlYzVmM2Y0OTU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDA2NTYxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNTNhNGRiZDk2ZTliYjA0Nzg2Y2NhNTJkNDgzZTdiYjM3NzJkYWVkOGY2ZmU2NzM2OWI4MDM4MjYyMGJhNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CC2BvLAbqwBdO-jvMaxc5YseoD7Z-oWI64C3_Mpv15Y) Why does the boss appear to be nan in the 4th round of training?
> 
> I have encountered the same problem. Have you solved it according to the author's method?

I recently came across the same issue while training / validating on my own dataset, the loss became NAN at about 80 epoches.However, the code works fine on the other dataset. Eventually I found that there is something wrong with my val dataset(the number of images is different with the number of labels) And I did not add the code "torch.nn.utils.clip_grad_norm_(net.parameters(), 1, norm_type=2)" So you may check your dataset.

**Comment by feng399-xinxin** [comment]
Hello, I have encountered the same problem. Have you resolved it？
